mono/metadata/threads.c

```diff
@@ -159,6 +159,18 @@ static MonoGHashTable *threads_starting_up = NULL;
 static GHashTable *joinable_threads;
 static gint32 joinable_thread_count;
 
+enum ThreadJoinableState {
```

Comment:
Please document the intended state machine for those 3 states. This makes it possible to validate your design separately from the code.

By that I mean something like this (mind that the example purposefully don't try to be correct):

The initial state is always NONE

When a thread detach the transition diagram is the following:
NONE -> PENDING (action XX happens)
PEDING -> ?? (action YY happens)
ADDED -> ERROR (this is an invalid transition because X, Y and Z)

Reply:
Will add that information.